app/controllers/api/v1/lookup_keys_controller.rb

```diff
@@ -11,9 +11,10 @@ class LookupKeysController < V1::BaseController
       param :per_page, String, :desc => "number of entries per request"
 
       def index
-        @lookup_keys = LookupKey.
-          authorized(:view_external_variables).
-          search_for(*search_options).paginate(paginate_options)
+        @lookup_keys = PuppetclassLookupKey.authorized(:view_external_variables).
```

Comment:
right now it works since the only thing you can search that isn't shared is puppetclass and paramclass.
paramclass id defined in both and puppetclass works on puppetclass_lookupkey probably since it exists through environment_classes.
It could become a problem so I would open a refactoring issue and maybe the bug in scoped_search will be fixed by then. Otherwise we can also put all the association in the parent which means the classes still aren't completely separated.